libs/ledger/src/consensus/consensus.cpp

```diff
@@ -27,20 +30,173 @@ using NextBlockPtr    = Consensus::NextBlockPtr;
 using Status          = Consensus::Status;
 using StakeManagerPtr = Consensus::StakeManagerPtr;
 
+namespace {
+using DRNG = fetch::random::LinearCongruentialGenerator;
+
+std::size_t SafeDecrement(std::size_t value, std::size_t decrement)
+{
+  if (decrement >= value)
+  {
+    return 0;
+  }
+  else
+  {
+    return value - decrement;
+  }
+}
+
+template <typename T>
+void DeterministicShuffle(T &container, uint64_t entropy)
+{
+  DRNG rng(entropy);
+  std::shuffle(container.begin(), container.end(), rng);
+}
+}  // namespace
+
 Consensus::Consensus(StakeManagerPtr stake, BeaconServicePtr beacon, MainChain const &chain,
-                     Identity mining_identity, uint64_t aeon_period, uint64_t max_committee_size)
+                     Identity mining_identity, uint64_t aeon_period, uint64_t max_committee_size,
+                     uint32_t block_interval_ms)
   : stake_{std::move(stake)}
   , beacon_{std::move(beacon)}
   , chain_{chain}
   , mining_identity_{std::move(mining_identity)}
   , mining_address_{Address(mining_identity_)}
   , aeon_period_{aeon_period}
   , max_committee_size_{max_committee_size}
+  , block_interval_ms_{block_interval_ms}
 {
   assert(stake_);
   FETCH_UNUSED(chain_);
 }
 
+Consensus::CommitteePtr Consensus::GetCommittee(Block const &previous)
+{
+  // Calculate the last relevant snapshot
+  uint64_t const last_snapshot =
+      previous.body.block_number - (previous.body.block_number % aeon_period_);
+
+  // Invalid to request a committee too far ahead in time
+  assert(committee_history_.find(last_snapshot) != committee_history_.end());
+
+  if (committee_history_.find(last_snapshot) == committee_history_.end())
+  {
+    FETCH_LOG_INFO(LOGGING_NAME,
+                   "No committee history found for block: ", previous.body.block_number, " AKA ",
+                   last_snapshot);
+    // TODO(jmw): Check: should return nullptr here?
```

Comment:
Yup, it looks like this is only used in consensus.cpp and it checks for a nullptr